### ISSUE TITLE
Ether in vehicle tanks display the wrong amount

### ISSUE THREAD

**Issue by gynther** [CleverRaven/Cataclysm-DDA#30562]
Installed a 60L tank in my vehicle and dropped some ether into it, the vehicle display then showed "1600,0L" of content in the tank.

I cannot craft hundreds of anesthetia kits so it seems it internally correct and just a display error.

Savegame:
[Royal Oak.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3189301/Royal.Oak.zip)

- OS: Windows 10 [64-bit] 
- Game Version: 0.D-3026-g510a637
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    DeadLeaves' Fictional Guns [FIC_Weapons],
    Extended Realistic Guns [realguns],
    Craftable Gun Pack [craftgp],
    More Survival Tools [More_Survival_Tools],
    Makeshift Items Mod [makeshift],
    Hydroponics [hydroponics],
    Simplified Nutrition [novitamins],
    Safe Autodoc [safeautodoc],
    Battery Migration for Existing Games [Battery_Overhaul_Legacy_Mode],
    Bens GF recipes [Tolerate_This],
    Alternative Map Key [alt_map_key]
]

![CDDA-ether-vehicletank](https://user-images.githubusercontent.com/622038/57892342-e06c2500-783e-11e9-9cbd-b974eacb7e27.jpg)

**Comment by neitsa** [comment]
I can definitely see the same bug on the latest version. A few notes about the bug below.

As for the display, it happens here in `veh_interact.cpp`:

https://github.com/CleverRaven/Cataclysm-DDA/blob/5b125138e8d89a56912554a6b57cc23d624c9b2d/src/veh_interact.cpp#L1289-L1299

Basically the content display for the ether tank value (1600L in this case) goes like this:

```
round_up( to_liter( pt.ammo_remaining() * stack ), 1 )
```

`ammo_remaining()` function:

https://github.com/CleverRaven/Cataclysm-DDA/blob/5b125138e8d89a56912554a6b57cc23d624c9b2d/src/vehicle_part.cpp#L211-L215

So if `contents` is not empty we get the number of charges in the tank, which is 0x1900 (6400):

```
>? pt.base.contents.back().charges
0x00001900
```
This value is then multiplied with `stack` which is given here:

```
auto stack = units::legacy_volume_factor / pt_ammo_cur->stack_size;
```

Therefore, `stack` is:

```
>? units::legacy_volume_factor
{value_=0x000000fa }
    value_: 0x000000fa
>? pt_ammo_cur->stack_size
0x00000001
```

* 0xfa / 1 = 0xfa = 250

Thus the content value is:

* 0x1900 * 0xfa = 6400 * 250 = 1600000

Which converted to liters gives 1600L.

Looking at the save file, searching for the tank (it's in `/save/Royal Oak/maps/3.4.0/109.137.0.map`):

```
{"id":"tank","base":{"typeid":"metal_tank","item_tags":["VEHICLE"],"contents":[{"typeid":"ether","charges":6400,"bday":2041591}]}
````

The number of charges is in the save file already... Since the tank is a 60L tank, it should not be able to contain more than 240 charges (240 * 250 = 60000ml = 60L)

**Comment by Jamuro-g** [comment]
tested with current experimental (11432)

- filled a rvs tank with 1 bottle of ether (0.75L)
result:
![image](https://user-images.githubusercontent.com/76928284/106952968-1eeff480-6732-11eb-965c-52e82d4aaa31.png)

The only "issue" i see is that the vehicle ui rounds to one decimal. Not a big deal, but that's why it shows 0.8 instead of 0.75

**Comment by anothersimulacrum** [comment]
Fixed in #42945
